astropy/io/ascii/ui.py

```diff
@@ -666,6 +668,26 @@ def write(table, output=None,  format=None, Writer=None, fast_writer=True, **kwa
 
     table = Table(table, names=kwargs.get('names'))
 
+    table0 = table[:0].copy()
+    core._apply_include_exclude_names(table0, kwargs.get('names'),
+                    kwargs.get('include_names'), kwargs.get('exclude_names'))
+    diff_format_with_names = set(kwargs.get('formats', [])) - set(table0.colnames)
+    same_include_names_with_exclude_names = set(kwargs.get('include_names', [])) & \
+                    set(kwargs.get('exclude_names', []))
+    diff_include_names_with_exclude_names = set(kwargs.get('include_names', [])) - \
+                    set(kwargs.get('exclude_names', []))
+
+    if diff_format_with_names:
+        warnings.warn(
+            '{} keys in the formats not to match the keys in names'
+            .format(diff_format_with_names), AstropyWarning)
+
+    if same_include_names_with_exclude_names:
```

Comment:
@taldcroft but previously you told me that to include ```include_names``` and ```exclude_names``` test cases and check if include_names = ['x','y'] and exclude_names=['y'] then show output 'x' is OK buy 'y' is not OK. so i added this. should i remove this ?